sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -1190,9 +1190,14 @@ def get_yielded_type(type_hint):
   if is_consistent_with(type_hint, Iterator[Any]):
     return type_hint.yielded_type
   if is_consistent_with(type_hint, Tuple[Any, ...]):
-    return Union[type_hint.tuple_types]
+    if isinstance(type_hint, TupleConstraint):
+      return Union[type_hint.tuple_types]
+    else:  # TupleSequenceConstraint
+      return type_hint.inner_type
   if is_consistent_with(type_hint, Iterable[Any]):
     return type_hint.inner_type
+  if is_consistent_with(type_hint, Dict[Any, Any]):
```

Comment:
Supporting this feels like a bug. Do we guard against it elsewhere?

Reply:
You're right, this branch has been removed.